src/nl/surf/eduhub_rio_mapper/api.clj

```diff
@@ -99,16 +99,9 @@
     (job-route (assoc-in request [:params :action] "dry-run-upsert")))
 
   (POST "/job/link/:rio-code/:type/:id"
-        {{:keys [type id rio-code]} :params :as request}
-    (when (#{"programs" "courses" "education-specifications"} type)
-      {:job (-> request
-                (select-keys [:institution-schac-home
-                              :institution-oin
-                              :trace-context])
-                (assoc :action "link"
-                       ::ooapi/type (types type)
-                       ::ooapi/id id
-                       ::rio/opleidingscode rio-code))}))
+        {{:keys [rio-code]} :params :as request}
+    (assoc (job-route (assoc-in request [:params :action] "link"))
+      ::rio/opleidingscode rio-code))
```

Comment:
Shouldn't this be the following (and why are there no test failing on this):

```clojure
(assoc-in (job-route (assoc-in request [:params :action] link))
          [:job ::rio/opleidinscode]
          rio-code)
```

Btw: proper indentation makes this more readable.